daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -393,7 +390,6 @@ class Engine(val config: EngineConfig = Engine.StableConfig) {
           handleError(err)
       }
 
-    @tailrec
```

Comment:
I'm not familiar enough with scala but won't this have an impact on performance?

Reply:
revererted

Reply:
again, I'm not familiar enough, but it seems that `loop` is now truly tail recursive in some cases, while in others it's recursive (but not tail recursive) by calling `interpretLoop`, so iiuc we'd lose stack safety

Reply:
@tailrec grantees that  direct recursive calls do not use stack. Taking into account we have one such call (for `getTime`) dropping the annotation has a potential impact. If there are a lot of successive `getTime` call in the Daml code, the interpreter stack may blow up. Very unlikely but at least theoretically possible.